operator/pkg/workqueuemetrics/metrics.go

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package workqueuemetrics
+
+import (
+	"github.com/cilium/cilium/pkg/metrics"
+	"github.com/cilium/cilium/pkg/metrics/metric"
+)
+
+const (
+	// LabelQueueName denotes which queue the metric is for
+	LabelQueueName = "queue_name"
+)
+
+func NewMetrics() *Metrics {
+	return &Metrics{
+		WorkQueueDepth: metric.NewGaugeVec(metric.GaugeOpts{
+			Namespace: metrics.CiliumOperatorNamespace,
+			Name:      "workqueue_depth",
+			Help:      "Current depth of the workqueue",
+		}, []string{LabelQueueName}),
+
+		WorkQueueAddsTotal: metric.NewCounterVec(metric.CounterOpts{
+			Namespace: metrics.CiliumOperatorNamespace,
+			Name:      "workqueue_adds_total",
+			Help:      "Total number of adds handled by the workqueue",
+		}, []string{LabelQueueName}),
+
+		WorkQueueLatency: metric.NewHistogramVec(metric.HistogramOpts{
+			Namespace: metrics.CiliumOperatorNamespace,
+			Name:      "workqueue_queue_duration_seconds",
+			Help:      "Duration in seconds an item stays in workqueue prior to request",
+			Buckets:   []float64{.01, .05, .1, .5, 1, 5, 10, 30, 60},
```

Comment:
Is it intentional that the buckets are different compared to how they were defined previously? I don't see any specific mention in the commit message. Same for the one below.

Reply:
You are right, I did update the buckets, I realized that the ones we had before with `prometheus.ExponentialBuckets(10e-9, 10, 10)` were not very useful, they are basically looking like this:
```
[]float64{
    1e-08,
    1e-07,
    1e-06,
    1e-05,
    0.0001,
    0.001,
    0.01,
    0.1,
    1.0,
    10.0,
}
```

Having larger values, especially for work which can take seconds, makes more sense.

I will document the change in the commit and the PR descriptions.